### PR TITLE
EASY-2674: Add Funding handler

### PR DESCRIPTION
**Pull request by lindareijnhoudt**
fixes EASY-2674

#### When applied it will
* Add tests
* Upgrade easy-emd in pom
* Add Funding handler

#### related pull requests on github
repo                       | PR                | note
-------------------------- | ----------------- | ---
easy-emd                      | [PR#22](https://github.com/DANS-KNAW/easy-emd/pull/22)     | 
easy-schema | [PR#82](https://github.com/DANS-KNAW/easy-schema/pull/82)

### CODE REVIEW

File: src/main/java/nl/knaw/dans/pf/language/ddm/handlers/FundingHandler.java [hunk 1]
```diff
@@ -0,0 +1,64 @@
...
+    author = new Author();
+    getTarget().getEmdContributor().getEasContributor().add(author);
+  }
+
+  @Override
+  protected void finishElement(final String uri, final String localName) throws SAXException {
+    if ("funding".equals(localName)) {
+      author.setRole(new Author.Role("Funder", "EASY"));
+      logger.debug("collected: " + author.toString());
+    }else {
```
Comment (rvanheest) [inline src/main/java/nl/knaw/dans/pf/language/ddm/handlers/FundingHandler.java:46]:
```suggestion
    }
    else {
```

File: src/main/java/nl/knaw/dans/pf/language/ddm/handlers/FundingHandler.java [hunk 2]
```diff
@@ -0,0 +1,64 @@
...
+  public void initFirstElement(final String uri, final String localName, final Attributes attributes) {
+    author = new Author();
+    getTarget().getEmdContributor().getEasContributor().add(author);
+  }
+
+  @Override
+  protected void finishElement(final String uri, final String localName) throws SAXException {
+    if ("funding".equals(localName)) {
+      author.setRole(new Author.Role("Funder", "EASY"));
+      logger.debug("collected: " + author.toString());
```
Comment (rvanheest) [inline src/main/java/nl/knaw/dans/pf/language/ddm/handlers/FundingHandler.java:45]: if you want to keep this debug log, please add some more context!
if not, please also remove the logger above 😉

File: src/test/java/nl/knaw/dans/pf/language/ddm/api/Ddm2EmdTest.java [hunk 3]
```diff
@@ -91,6 +91,7 @@ private static boolean canConnect(String url) {
             return true;
         }
         catch (IOException e) {
+            System.out.println(e.getMessage());
```
Comment (rvanheest) [inline src/test/java/nl/knaw/dans/pf/language/ddm/api/Ddm2EmdTest.java:94]:
```suggestion
```

File: src/main/java/nl/knaw/dans/pf/language/ddm/handlers/FundingHandler.java [hunk 4]
```diff
@@ -0,0 +1,60 @@
...
+import nl.knaw.dans.pf.language.emd.types.Author;
+import nl.knaw.dans.pf.language.emd.types.EntityId;
+import nl.knaw.dans.pf.language.xml.crosswalk.CrosswalkHandler;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.xml.sax.Attributes;
+import org.xml.sax.SAXException;
+
+public class FundingHandler  extends CrosswalkHandler<EasyMetadata> {
+  private static final Logger logger = LoggerFactory.getLogger(FundingHandler.class);
```
Comment (rvanheest) [inline src/main/java/nl/knaw/dans/pf/language/ddm/handlers/FundingHandler.java:28]:
```suggestion
```

File: src/main/java/nl/knaw/dans/pf/language/ddm/handlers/FundingHandler.java [hunk 5]
```diff
@@ -0,0 +1,60 @@
...
+ * limitations under the License.
+ */
+package nl.knaw.dans.pf.language.ddm.handlers;
+
+import nl.knaw.dans.pf.language.emd.EasyMetadata;
+import nl.knaw.dans.pf.language.emd.types.Author;
+import nl.knaw.dans.pf.language.emd.types.EntityId;
+import nl.knaw.dans.pf.language.xml.crosswalk.CrosswalkHandler;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
```
Comment (rvanheest) [inline src/main/java/nl/knaw/dans/pf/language/ddm/handlers/FundingHandler.java:23]:
```suggestion
```

File: src/main/java/nl/knaw/dans/pf/language/ddm/handlers/FundingHandler.java [hunk 6]
```diff
@@ -0,0 +1,60 @@
...
+import org.xml.sax.Attributes;
+import org.xml.sax.SAXException;
+
+public class FundingHandler  extends CrosswalkHandler<EasyMetadata> {
```
Comment (rvanheest) [inline src/main/java/nl/knaw/dans/pf/language/ddm/handlers/FundingHandler.java:27]:
```suggestion
public class FundingHandler extends CrosswalkHandler<EasyMetadata> {
```

File: src/test/java/nl/knaw/dans/pf/language/ddm/api/Ddm2EmdTest.java [hunk 7]
```diff
@@ -91,6 +91,7 @@ private static boolean canConnect(String url) {
             return true;
         }
         catch (IOException e) {
+//            System.out.println(e.getMessage());
```
Comment (rvanheest) [inline src/test/java/nl/knaw/dans/pf/language/ddm/api/Ddm2EmdTest.java:94]:
```suggestion
```

**Comment by lindareijnhoudt** [comment]
i don't see how that would improve the testing, it is already covered in the 'funding.xml' and emd-actual-expected.xml in xml-format, and in the FundingTest it is done on the POJOs

**Pull request merged**